### PR TITLE
Login endpoint and token verifying

### PR DESCRIPTION
**Pull request by eduardpeters**
Adding login endpoint logic:
Check if user exists
Compare password hashes using bcrypt
Issue a JSON Web Token with payload
Adding verifyJWT middleware to check user authorization
Adding a get request to login endpoint to test this authorization if needed

### CODE REVIEW

**Review by laurapellicerrahan — APPROVED** [review]
Perfect!

**Review by RLANFRAN86 — COMMENTED** [review]
review ok

File: controllers/loginController.js [hunk 1]
```diff
@@ -1,7 +1,61 @@
...
+    if (result) {
+        return ("Client");
+    }
+    result = await DeliverersManager.getDelivererByUuid(userUuid);
+    if (result) {
+        return ("Delivery");
+    }
+    result = await AdminsManager.getAdminByUuid(userUuid);
+    if (result) {
+        return ("admin");
```
Comment (Rotosama) [inline controllers/loginController.js:19]: admin => Admin

**Review by Rotosama — APPROVED** [review]
great job!!

**Pull request merged**